### PR TITLE
PWGHF: Developements on Charm Resonances data creator

### PR DESCRIPTION
**Pull request by Luca610**
This PR introduces some modifications to the charm resonances data creator to reconstruct on-the-fly V0s daughters to be paired with D mesons, in order to reduce the memory consumption of the data creation workflow.

### CODE REVIEW

File: PWGHF/D2H/TableProducer/dataCreatorCharmResoReduced.cxx [hunk 1]
```diff
@@ -134,14 +138,35 @@ struct HfDataCreatorCharmResoReduced {
...
+    float mLambda;
+    uint8_t v0Type;
+  } candidateV0;
+
   using CandsDplusFiltered = soa::Filtered<soa::Join<aod::HfCand3Prong, aod::HfSelDplusToPiKPi>>;
   using CandsDplusFilteredWithMl = soa::Filtered<soa::Join<aod::HfCand3Prong, aod::HfSelDplusToPiKPi, aod::HfMlDplusToPiKPi>>;
   using CandDstarFiltered = soa::Filtered<soa::Join<aod::HfD0FromDstar, aod::HfCandDstars, aod::HfSelDstarToD0Pi>>;
   using CandDstarFilteredWithMl = soa::Filtered<soa::Join<aod::HfD0FromDstar, aod::HfCandDstars, aod::HfSelDstarToD0Pi, aod::HfMlDstarToD0Pi>>;
   using TracksWithPID = soa::Join<aod::Tracks, aod::TracksCov, aod::TracksExtra, aod::TrackSelection, aod::TracksDCA, aod::pidTPCFullPi, aod::pidTOFFullPi, aod::pidTPCFullKa, aod::pidTOFFullKa, aod::pidTPCFullPr, aod::pidTOFFullPr>;
+  using TracksIUWithPID = soa::Join<aod::TracksIU, aod::TracksExtra, aod::TracksCovIU, aod::pidTPCPi, aod::pidTPCPr>;
```
Comment (fgrosa) [inline PWGHF/D2H/TableProducer/dataCreatorCharmResoReduced.cxx:169]:
```suggestion
  using TracksIUWithPID = soa::Join<aod::TracksIU, aod::TracksExtra, aod::TracksCovIU, aod::pidTPCPi, aod::pidTPCPr>;
```
I would add the possibility to apply a cut also on the NsigmaTOF (applied in the trigger for the proton from Lambda), so we need to consume the TOF tables

File: PWGHF/D2H/TableProducer/dataCreatorCharmResoReduced.cxx [hunk 2]
```diff
@@ -201,59 +225,183 @@ struct HfDataCreatorCharmResoReduced {
...
     }
-    if (std::fabs(v0.mLambda() - MassLambda0) > cfgV0Cuts.deltaMassLambda) {
-      CLRBIT(selMap, Lambda);
+    if (std::fabs(candidateV0.mLambda - MassLambda0) > cfgV0Cuts.deltaMassLambda) {
+      CLRBIT(candidateV0.v0Type, Lambda);
+      CLRBIT(candidateV0.v0Type, AntiLambda);
     }
-    if (std::fabs(v0.mAntiLambda() - MassLambda0) > cfgV0Cuts.deltaMassLambda) {
-      CLRBIT(selMap, AntiLambda);
+    // PID
```
Comment (fgrosa) [inline PWGHF/D2H/TableProducer/dataCreatorCharmResoReduced.cxx:385]: Add also TOF

**Review by fgrosa — CHANGES_REQUESTED** [review]
Ciao @Luca610, many thanks for the nice work! I have a single comment, namely that I would add the possibility to apply also a cut on the NsigmaTOF for the V0 daughters (only if TOF is present) to be able to reproduce the cut applied in the software triggers.

File: PWGHF/D2H/TableProducer/dataCreatorCharmResoReduced.cxx (hunk 1 above)
Comment (Luca610) [reply PWGHF/D2H/TableProducer/dataCreatorCharmResoReduced.cxx]: Done!

**Comment by Luca610** [comment]
Hi @fgrosa, thank you for your comments! With my latest commit I introduced also a selection on nSigmaTOF only for the protons coming from Lambdas as suggested!

**Review by fgrosa — APPROVED** [review]

**Pull request merged**